### PR TITLE
docs: Add documentation about `parameters.snap`

### PR DESCRIPTION
**Pull request by akashin**
Fixes https://github.com/near/nearcore/issues/7642.

### CODE REVIEW

File: docs/architecture/gas/parameter_definition.md [hunk 1]
```diff
@@ -1,13 +1,14 @@
 # Parameter Definitions
 
 Gas parameters are a subset of runtime parameters that are defined in
-[core/primitives/res/runtime_configs/parameters.yaml](https://github.com/near/nearcore/blob/d0dc37bf81f7e7bde9c560403b085fae04108659/core/primitives/res/runtime_configs/parameters.yaml).
+[core/primitives/res/runtime_configs/parameters.yaml](https://github.com/near/nearcore/blob/master/core/primitives/res/runtime_configs/parameters.yaml).
```
Comment (akashin) [inline docs/architecture/gas/parameter_definition.md:4]: I'm confused whether we should use a link to the version at specific commit or at `master` here.
In my understanding, we use specific commit to make sure documentation doesn't rot when the file is moved. But for some reason, the previous link turned out to be invalid, so this rule got violated.

Maybe instead we should always point to `master` in links to a file and have some linter that checks for broken links in documentation?

Comment (jakmeier) [reply docs/architecture/gas/parameter_definition.md:4]: Yeah, that's weird, that's exactly why permalinks exists^^

Probably you used a commit hash of a PR branch which you then deleted? Then this happens. If you use a permalink of the master branch, I don't think this should happen.

But in this instance, I think linking to a branch is preferred anyway, as we don't want to link to outdated parameter values. (It's easy to not notice that one looks at a 2 years old version of the file.)

**Review by jakmeier — APPROVED** [review]
Thanks for updating the docs!

File: docs/architecture/gas/parameter_definition.md (hunk 1 above)
Comment (akashin) [reply docs/architecture/gas/parameter_definition.md:4]: Yes, that's my thoughts as well, will use a link to `master` branch here.

**Pull request merged**